src/code/bank0.asm

```diff
@@ -7090,125 +7130,131 @@ UpdateIndoorRoomStatus::
-    jp   Func_354B                                ;; 00:373C $C3 $4B $35
+    call ObjectPositionToRoomObjectAddress        ;; 00:3733 $CD $EE $35
+    ld   bc, HorizontalObjectOffsets              ;; 00:3736 $01 $E1 $37
+    ld   de, BossDoorObjectIds                    ;; 00:3739 $11 $24 $37
+    jp   CopyIndoorsMacroObjectsToRoom            ;; 00:373C $C3 $4B $35
 
-func_373F::
+; Make lists of the positions of current doors in the room
+MakeListOfDoorPositions::
     ld   d, $00                                   ;; 00:373F $16 $00
     ld   hl, wC1F0                                ;; 00:3741 $21 $F0 $C1
     add  hl, de                                   ;; 00:3744 $19
+    ; Decrement bc to be at object position
     dec  bc                                       ;; 00:3745 $0B
     ld   a, [bc]                                  ;; 00:3746 $0A
+    ; Put position in scratch space
     ld   [hl], a                                  ;; 00:3747 $77
     push af                                       ;; 00:3748 $F5
+    ; Put Y position in scratch space
     and  $F0                                      ;; 00:3749 $E6 $F0
     ld   hl, wC1E0                                ;; 00:374B $21 $E0 $C1
     add  hl, de                                   ;; 00:374E $19
     ld   [hl], a                                  ;; 00:374F $77
+    ; Put X position in scratch space
     pop  af                                       ;; 00:3750 $F1
     swap a                                        ;; 00:3751 $CB $37
     and  $F0                                      ;; 00:3753 $E6 $F0
     ld   hl, wC1D0                                ;; 00:3755 $21 $D0 $C1
     add  hl, de                                   ;; 00:3758 $19
     ld   [hl], a                                  ;; 00:3759 $77
+    ; Increment bc to be back at object type
     inc  bc                                       ;; 00:375A $03
     ret                                           ;; 00:375B $C9
 
-data_375C::
+StairsDoorObjectIds::
     db   $AF, $B0                                 ;; 00:375C
 
 LoadObject_StairsDoor::
-    push bc                                       ; $375E: $C5 ;; 00:375E $C5
-    call label_35EE                               ; $375F: $CD $EE $35 ;; 00:375F $CD $EE $35
-    ld   bc, data_37E4                            ; $3762: $01 $E4 $37 ;; 00:3762 $01 $E4 $37
-    ld   de, data_375C                            ; $3765: $11 $5C $37 ;; 00:3765 $11 $5C $37
-    jp   Func_354B                                ; $3768: $C3 $4B $35 ;; 00:3768 $C3 $4B $35
+    push bc                                       ;; 00:375E $C5
+    call ObjectPositionToRoomObjectAddress        ;; 00:375F $CD $EE $35
+    ld   bc, VerticalObjectOffsets                ;; 00:3762 $01 $E4 $37
+    ld   de, StairsDoorObjectIds                  ;; 00:3765 $11 $5C $37
+    jp   CopyIndoorsMacroObjectsToRoom            ;; 00:3768 $C3 $4B $35
 
-data_376B::
+FlipWallObjectIds::
```

Comment:
We use a `RevolvingDoor` wording in some places already. Is a `FlipWall` the same thing?

Reply:
Honestly I don't know, I just called it that because it's used in the existing subroutine LoadObject_FlipWall directly below, which I didn't name. I can try to find out if the FlipWall and the RevolvingDoor is the same thing (I assume it is, but there's also LoadObject_OneWayArrow which I'm also not sure what is).

Reply:
Alright, let's leave it like this for now then 👍

Reply:
@kemenaran I decided to rename FlipWall to RevolvingDoor after all :) OneWayArrow is the opposite side of the RevolvingDoor which can't be entered, so I guess that can stay as-is.